api/utility/request_handler.sh

```diff
@@ -7,16 +7,77 @@
 # Globals:
 #   None
 # Arguments:
-#   The response body.
+#   A JSON object with keys `response_body` and `http_code` (created by makeFullResponse())
 # Outputs:
 #   Returns the error if there is an error.
 request_error() {
-  error_code=$(echo "$1" | jq '.code' )
-  if [ "$error_code" == "null" ]; then
-    return
+  local body=$(getBodyFromFullResponse "$1")
+  local http_code=$(getHttpStatusFromFullResponse "$1")
+
+  local endpointName="$2"
+  local non200=0
+  local hasResponseErr=0
+
+  if [[ $http_code -lt 200 || $http_code -ge 300 ]]; then
+    non200=1
+    echo "Error: ${endpointName} request failed with status $http_code" >&2
+  fi
+  
+  error_code=$(echo "$body" | jq '.code' )
+  if [[ -n "$error_code" && "$error_code" != "null" ]]; then
+    hasResponseErr=1
+    echo "$body" | jq .
   fi
 
-  echo "$1" | jq .
+  if [[ $hasResponseErr -eq 1 || $non200 -eq 1 ]]; then
+    exit 1
+  fi
+}
 
-  exit 0
+#######################################
+# Gets response body from makeFullResponse() json object
+# Globals:
+#   None
+# Arguments:
+#   A JSON object with keys `response_body` and `http_code` (created by makeFullResponse())
+# Outputs:
+#   A JSON object with the response body
+getBodyFromFullResponse() {
+  local full_response=$1
+  echo "$full_response" | jq ".response_body  // empty"
 }
+
+#######################################
+# Gets response body from makeFullResponse() json object
+# Globals:
+#   None
+# Arguments:
+#   A JSON object with keys `response_body` and `http_code` (created by makeFullResponse())
+# Outputs:
+#   A string with the http status code of the response
+getHttpStatusFromFullResponse() {
+  local full_response=$1
+  echo "$full_response" | jq ".http_code"
+}
+
+#######################################
+# Combines the HTTP response code and response body into a json object
+# Globals:
+#   None
+# Arguments:
+#   The response body.
+#   The http code (int)
+# Outputs:
+#   A JSON object with format {"http_code": <int>, "response_body": <string>}
+makeFullResponse() {
+  local body=$1
```

Comment:
Is there any reason why you use the body as the first parameter? It seems to be more logical (at least for me) to list the optional parameters at the end of the parameter list.

Reply:
Yes, there is a reason. A bad one. When I originally wrote the function I thought both inputs would be required. Thanks for the suggestion. Updating